Dropped unsupported alpine 3.18 and soon unsupported alpine 3.19, add…

alaendle opened the pull request:
…ed alpine 3.21 and 3.22.  Also dropped unsupported node 18 and added node 22. No other changes.

alaendle committed:
Dropped unsupported alpine 3.18 and soon unsupported alpine 3.19, added alpine 3.21 and 3.22.  Also dropped unsupported node 18 and added node 22. No other changes.

alaendle commented:
Ultimately, I'm very interested in getting an up-to-date image on the Docker registry (https://hub.docker.com/_/mongo-express). But instead of an issue, I wanted to create a PR right away; maybe that will shorten the wait. Please let me know if there's anything else I can do to contribute to this goal.

rtritto commented:
@alaendle thanks for the PR!

Please can you update?

https://github.com/mongo-express/mongo-express-docker/blob/075234363b3c008c7c2e97de8a3639e23a75cfcd/README.md?plain=1#L32

```diff
- 1.0/20-alpine3.18
+ 1.0/20-alpine3.20
```

---

For the Docker Image version, I think you can help/contribute, I created #133

alaendle committed:
Keep ordering consistent and updated readme.

rtritto commented:
LGTM
Before merge this PR, we need to update the Node version in `mongo-express`

shakaran committed:
Refresh to Node 22/24 and Alpine 3.22/3.24

The pull request has been open since October and the versions it introduced
have since aged out:

  Node 20      EOL 2026-04-30
  Alpine 3.21  EOL 2026-11-01, three months away

Merging as-is would ship two variants on an end-of-life runtime and two more
needing replacement before the year is out. Regenerated with versions.sh and
apply-templates.sh, so the variants become 22/24-alpine3.22/3.24, the nearest
expiry being Alpine 3.22 in May 2027.

This also lines up with mongo-express/mongo-express#1894, which moved the main
repository's own Dockerfile to node:24-alpine3.24.

Verified rather than assumed, since pairing the 2024-era 1.0 branch with Node 24
is the risky part: 1.0/24-alpine3.24 builds, and the image runs Node v24.18.1
with the entrypoint behaving normally.

shakaran reviewed (approved):
Approving on the maintainers' instruction, with one thing on the record: **this branch contains a commit of mine** (`dec1b21`), the refresh to Node 22/24 and Alpine 3.22/3.24. So this is not an independent review of the whole branch — @alaendle's two commits are, mine is not.

What the branch does:

- drops Node 18 and Alpine 3.18/3.19, all end-of-life — @alaendle's original change, and overdue
- refreshed from Node 20 / Alpine 3.21, which aged out while this sat in the queue (Node 20 EOL 2026-04-30, Alpine 3.21 EOL 2026-11-01)
- final variants `22/24-alpine3.22/3.24`, nearest expiry May 2027
- regenerated through `versions.sh` and `apply-templates.sh`, nothing hand-edited
- README folder reference updated, it pointed at a variant that no longer exists

Verified rather than assumed, since pairing the 2024-era 1.0 branch with Node 24 was the plausible breakage: `1.0/24-alpine3.24` builds clean and the image runs `v24.18.1` with the entrypoint behaving normally.

Merging with a merge commit to match this repository's history and keep @alaendle's commits distinct.
